`opa build` panics on second end-of-line comment in function call

<!-- Thanks for opening an issue to request a feature or file a bug!
If you provide some basic information it helps us address problems faster. -->

## Expected Behavior

The following rego code should compile, run _and `opa build`_.

`testcase.rego`:

```rego
package testcase

rule1 = contains(
    "", # first comment
    "", # second comment lets `opa build` panic
)
```

## Actual Behavior

`opa build` panics:

```
$ opa build testcase.rego 
panic: overwriting non-nil beforeEnd

goroutine 1 [running]:
github.com/open-policy-agent/opa/format.(*writer).beforeLineEnd(...)
	/src/format/format.go:1044
github.com/open-policy-agent/opa/format.(*writer).insertComments(0xc000114c00, {0xc00011e500, 0x1426700, 0xc00021f1b8}, 0xc0001473c0)
	/src/format/format.go:378 +0x117
github.com/open-policy-agent/opa/format.(*writer).writeTermParens(0xc000114c00, 0x3a, 0xc0004192f0, {0xc00011e500, 0x412afd, 0xc00011e4b0})
	/src/format/format.go:517 +0x57
github.com/open-policy-agent/opa/format.(*writer).writeTerm(...)
	/src/format/format.go:513
github.com/open-policy-agent/opa/format.(*writer).writeFunctionCallPlain(0xc000114c00, {0xc000419278, 0x3, 0x8}, {0xc000412e50, 0x2, 0x2})
	/src/format/format.go:498 +0x1e6
github.com/open-policy-agent/opa/format.(*writer).writeCall(0xc000114c00, 0x0, {0xc000419278, 0x3, 0x3}, {0xc000412e50, 0x2, 0x2})
	/src/format/format.go:602 +0x26d
github.com/open-policy-agent/opa/format.(*writer).writeTermParens(0xc000114c00, 0x54, 0xc000419260, {0xc000412e40, 0x13eb120, 0xc000412e40})
	/src/format/format.go:548 +0x405
github.com/open-policy-agent/opa/format.(*writer).writeTerm(...)
	/src/format/format.go:513
github.com/open-policy-agent/opa/format.(*writer).writeHead(0xc000114c00, 0xc00017e460, 0x0, 0x1, {0xc000412e40, 0x2, 0x2})
	/src/format/format.go:366 +0x312
github.com/open-policy-agent/opa/format.(*writer).writeRule(0xc000114c00, 0xc000147580, 0x0, {0xc000412e40, 0x2, 0x2})
	/src/format/format.go:248 +0x20a
github.com/open-policy-agent/opa/format.(*writer).writeRules(0xc000412de0, {0xc00011e4e8, 0x1, 0x0}, {0xc000412e30, 0x0, 0xc00021f5f8})
	/src/format/format.go:223 +0x72
github.com/open-policy-agent/opa/format.(*writer).writeModule(0xc000114c00, 0xc0004021c0)
	/src/format/format.go:190 +0x2d3
github.com/open-policy-agent/opa/format.Ast({0xc75320, 0xc000402150})
	/src/format/format.go:74 +0x193
github.com/open-policy-agent/opa/format.Source({0x7fff4a746288, 0xd}, {0xc000174200, 0xffffffffffffffff, 0x4a2c91})
	/src/format/format.go:25 +0x7d
github.com/open-policy-agent/opa/bundle.(*Bundle).FormatModules(0xc00014a580, 0x0)
	/src/bundle/bundle.go:778 +0xeb
github.com/open-policy-agent/opa/compile.(*Compiler).Build(0xc00021fb98, {0xe3a790, 0xc0000340c0})
	/src/compile/compile.go:200 +0x11f
github.com/open-policy-agent/opa/cmd.dobuild({0xc000130168, 0xc000118480, 0x0, 0x0, {{0x0, 0x0, 0x0}, 0x0}, {0xd0f294, 0xd}, ...}, ...)
	/src/cmd/build.go:293 +0x70d
github.com/open-policy-agent/opa/cmd.init.1.func2(0xc00013c280, {0xc00011c590, 0x1, 0x1})
	/src/cmd/build.go:210 +0x6c
github.com/spf13/cobra.(*Command).execute(0xc00013c280, {0xc00011c560, 0x1, 0x1})
	/src/vendor/github.com/spf13/cobra/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0x13ddb80)
	/src/vendor/github.com/spf13/cobra/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/src/vendor/github.com/spf13/cobra/command.go:902
main.main()
	/src/main.go:15 +0x25
```

However, the code compiles and runs:

```
$ opa eval -d testcase.rego 'data.testcase.rule1'
{
  "result": [
    {
      "expressions": [
        {
          "value": true,
          "text": "data.testcase.rule1",
          "location": {
            "row": 1,
            "col": 1
          }
        }
      ]
    }
  ]
}
```

## Steps to Reproduce the Problem

Save `testcase.rego` to disk and run the commands shown above.

* OPA version: 0.32.1


<!--
If this is a bug report please provide as much detail as possible so that we can
reproduce the problem. Examples:

* OPA version
* Example query, input, data, and policy that OPA was given
* Example output that OPA returned
* For server and CLI, the flags/configuration that you provided to OPA
* For server, any relevant log messages from OPA
* For Go and Wasm, the arguments you invoked OPA with
-->

## Additional Info

<!--
Any additional information you think might be helpful. Examples include the environment
where OPA was running (e.g., if inside Kubernetes, what resource limits did you configure
OPA with?), how long OPA had been running for, what was happening around the time
when you identified the problem, etc.
-->

--- a/format/format.go
+++ b/format/format.go
@@ -490,15 +490,13 @@ func (w *writer) writeFunctionCall(expr *ast.Expr, comments []*ast.Comment) []*a
 
 func (w *writer) writeFunctionCallPlain(terms []*ast.Term, comments []*ast.Comment) []*ast.Comment {
 	w.write(terms[0].String() + "(")
-	if len(terms) > 1 {
-		for _, v := range terms[1 : len(terms)-1] {
-			comments = w.writeTerm(v, comments)
-			w.write(", ")
-		}
-		comments = w.writeTerm(terms[len(terms)-1], comments)
+	defer w.write(")")
+	args := make([]interface{}, len(terms)-1)
+	for i, t := range terms[1:] {
+		args[i] = t
 	}
-	w.write(")")
-	return comments
+	loc := terms[0].Location
+	return w.writeIterable(args, loc, closingLoc(0, 0, '(', ')', loc), comments, w.listWriter())
 }
 
 func (w *writer) writeWith(with *ast.With, comments []*ast.Comment) []*ast.Comment {

--- a/format/testfiles/test.rego.formatted
+++ b/format/testfiles/test.rego.formatted
@@ -98,7 +98,10 @@ fn2(
 	split(x, y, c)
 	trim(a, z, d) # function comment 1
 	split(c[0], d, b)
-	x = sprintf("hello %v", ["world"])
+	x = sprintf(
+		"hello %v",
+		["world"],
+	)
 	#function comment 2
 } # function comment 3
 

--- a/format/testfiles/test_fun_args_with_linebreaks.rego
+++ b/format/testfiles/test_fun_args_with_linebreaks.rego
@@ -1,0 +1,16 @@
+package test
+
+p {
+    x := count(
+        [1, 2, 3] # four
+    )
+    y := concat(
+        "/",
+        ["foo", "bar"],
+    )
+    z := concat("/",
+        [
+            "foo",
+            "bar",
+        ])
+}

--- a/format/testfiles/test_fun_args_with_linebreaks.rego.formatted
+++ b/format/testfiles/test_fun_args_with_linebreaks.rego.formatted
@@ -1,0 +1,18 @@
+package test
+
+p {
+	x := count([1, 2, 3]) # four
+
+	y := concat(
+		"/",
+		["foo", "bar"],
+	)
+
+	z := concat(
+		"/",
+		[
+			"foo",
+			"bar",
+		],
+	)
+}

--- a/format/testfiles/test_issue_3836.rego
+++ b/format/testfiles/test_issue_3836.rego
@@ -1,0 +1,6 @@
+package testcase
+
+rule1 = contains(
+    "", # first comment
+    "", # second comment
+)

--- a/format/testfiles/test_issue_3836.rego.formatted
+++ b/format/testfiles/test_issue_3836.rego.formatted
@@ -1,0 +1,6 @@
+package testcase
+
+rule1 = contains(
+	"", # first comment
+	"", # second comment
+)

